Upgraded the TUtoolbox to 2023.0.0.666

chbwien opened the pull request:
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/94711)

chbwien committed:
Upgraded the TUtoolbox to 2023.0.0.666

chbwien commented:
That's odd. The message says:
Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?
And the checker says, only 1.2.0 is supported....

chbwien committed:
Downgraded the schema to 1.2.0 as requested by the automatic checker

mdanish-kh commented on manifests/t/TUitTUWien/TUtoolbox/2023.0.0.666/TUitTUWien.TUtoolbox.locale.de-DE.yaml:
```diff
@@ -0,0 +1,24 @@
+# CopyrightUrl: 
+ShortDescription: Ein Programm zur Installation aller benötigten Werkzeuge und Patches für TU-Mitglieder.
+ReleaseNotesUrl: https://service.it.tuwien.ac.at/swsupport/tools/tutoolbox/index.html#history
+Moniker: tutoolbox
```
```suggestion
```
Only the `ManifestType: defaultLocale` supports Moniker field

chbwien replied:
changed it with the current push

mdanish-kh commented:
> That's odd. The message says: Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)? And the checker says, only 1.2.0 is supported....

That's a mistake, please ignore that message. The PR template was modified a bit too early.

mdanish-kh commented on manifests/t/TUitTUWien/TUtoolbox/2023.0.0.666/TUitTUWien.TUtoolbox.installer.yaml:
```diff
@@ -0,0 +1,22 @@
+Scope: machine
+UpgradeBehavior: install
+Installers:
+  - Architecture: neutral
```
```suggestion
  - Architecture: x86
```
Neutral includes all architectures including arm and arm64

chbwien replied:
That's right. And it works tested on them. So need to change.

mdanish-kh replied:
@chbwien We've actually move away from using `Neutral` as an architecture type. Best practice is that if it works on all platforms, then you should create separate Installer nodes for them as such

```yaml
PackageIdentifier: TUitTUWien.TUtoolbox
PackageVersion: 2023.0.0.666
InstallerType: inno
InstallerSwitches:
  Custom: /suppressmsgboxes
MinimumOSVersion: 10.0.0.0
Scope: machine
Platform:
  - Windows.Desktop
UpgradeBehavior: install
Installers:
- Architecture: x64
  InstallerUrl: https://service.it.tuwien.ac.at/swsupport/tools/tutoolbox/versions/TUToolBox_2023_0_0_666.exe
  InstallerSha256: 64bc1e36cc2d980bab084980cf96581ab6b825f9dbee41954f1179e615b01d18
- Architecture: x86
  InstallerUrl: https://service.it.tuwien.ac.at/swsupport/tools/tutoolbox/versions/TUToolBox_2023_0_0_666.exe
  InstallerSha256: 64bc1e36cc2d980bab084980cf96581ab6b825f9dbee41954f1179e615b01d18
- Architecture: arm
  InstallerUrl: https://service.it.tuwien.ac.at/swsupport/tools/tutoolbox/versions/TUToolBox_2023_0_0_666.exe
  InstallerSha256: 64bc1e36cc2d980bab084980cf96581ab6b825f9dbee41954f1179e615b01d18
- Architecture: arm64
  InstallerUrl: https://service.it.tuwien.ac.at/swsupport/tools/tutoolbox/versions/TUToolBox_2023_0_0_666.exe
  InstallerSha256: 64bc1e36cc2d980bab084980cf96581ab6b825f9dbee41954f1179e615b01d18
ManifestType: installer
ManifestVersion: 1.2.0
```
You can use this as your installer manifest. I've moved the common fields outside as they get inherited in each installer node. Also you don't need to pass Silent and SilentWithProgress switches as they get automatically passed to installer when installer type is set to inno.

chbwien committed:
Removed the moniker field in the german locale

mdanish-kh reviewed (changes requested):
https://github.com/microsoft/winget-pkgs/pull/94711#discussion_r1088160460

chbwien committed:
Changed the architecture section as proposed

I changed the architecture section as proposed, BUT I had to include the /silent parameters, because /supressmsgboxes should not be used when interactive. And when only setting the parameter for silent, I have to include it, because the default is removed (so silent is missing). Just checked.

stephengillie commented:
/azp run
